### ISSUE TITLE
Atomics/wake/undefined-index-defaults-to-zero.js fails intermittently

### ISSUE THREAD

**Issue by anba** [tc39/test262#1529]
https://github.com/tc39/test262/blob/master/test/built-ins/Atomics/wake/undefined-index-defaults-to-zero.js fails intermittently when the test system is under pressure, for example I can reliably reproduce the failure when I run the test in 16 tmux panes simultaneously.

**Comment by rwaldron** [comment]
I will start by attempting to simplify this test case and see where it takes me.

**Comment by leobalter** [comment]
@anba can you confirm if the problem persists? I believe it's fixed by #1533

**Comment by anba** [comment]
Yes, it's still an issue.

FWIW I'd probably rewrite the test as follows to be more resistant against spurious failures:

```js
var WAIT_INDEX = 0;
var STARTED_INDEX = 1;
var REPORTED_INDEX = 2;

// Large timeout value to avoid spurious failures on slow machines resp. when
// the system is under pressure (for example in CI environments).
var TIMEOUT = 10000;

var WORKER_COUNT = 2;

for (var i = 0; i < WORKER_COUNT; i++) {
  $262.agent.start(`
    $262.agent.receiveBroadcast(function(sab) {
      const i32a = new Int32Array(sab);
      Atomics.add(i32a, ${STARTED_INDEX}, 1);
      $262.agent.report(Atomics.wait(i32a, ${WAIT_INDEX}, 0, ${TIMEOUT}));
      Atomics.add(i32a, ${REPORTED_INDEX}, 1);
      $262.agent.leaving();
    });
  `);
}

const i32a = new Int32Array(
  new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT * 4)
);

$262.agent.broadcast(i32a.buffer);


// Wait until both workers started.
$262.agent.waitUntil(i32a, STARTED_INDEX, WORKER_COUNT);


// Wake at index 0, undefined => 0.
var woken = 0;
while ((woken = Atomics.wake(i32a, undefined, 1)) === 0) ;
assert.sameValue(woken, 1, 'Atomics.wake(i32a, undefined, 1) returns 1');

// Wait until report was delivered.
while (Atomics.compareExchange(i32a, REPORTED_INDEX, 1, 0) !== 1) ;
assert.sameValue($262.agent.getReport(), 'ok', '$262.agent.getReport() returns "ok"');


// Wake again at index 0, default => 0.
var woken = 0;
while ((woken = Atomics.wake(i32a, /*, default values used */)) === 0) ;
assert.sameValue(woken, 1, 'Atomics.wake(i32a /*, default values used */) returns 1');

// Wait until report was delivered.
while (Atomics.compareExchange(i32a, REPORTED_INDEX, 1, 0) !== 1) ;
assert.sameValue($262.agent.getReport(), 'ok', '$262.agent.getReport() returns "ok"');

```
